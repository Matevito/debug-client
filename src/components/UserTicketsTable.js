import React from 'react'

import { Link } from "react-router-dom";
import {
    Box,
    Paper,
    Table,
    TableContainer,
    TableHead,
    TableBody,
    TableRow,
    TableCell,
    IconButton,
    Typography,
    Button,
} from "@mui/material";
import BuildIcon from '@mui/icons-material/Build';

const colorHigh = "#cf6d5b";
const colorMid = " #ffc15c";
const colorLow = " #b8f8e7";

const Row = ({ issue, project}) => {
    let issueColor;
    if (issue.priority === "low"){
        issueColor = colorLow
    } else if (issue.priority === "mid"){
        issueColor = colorMid
    } else if (issue.priority === "high"){
        issueColor = colorHigh
    };
    return (
        <TableRow>
            <TableCell>
                {issue.title}
            </TableCell>

            <TableCell sx={{ backgroundColor: issueColor}} align="center">
                <Typography>{issue.priority}</Typography>
            </TableCell>

            <TableCell>{issue.type}</TableCell>

            <TableCell>
                <Link
                    to={`/project/${project._id}`}
                    style={{ color: "inherit", textDecoration: "none"}} 
                >
                    {project.title}
                </Link>
            </TableCell>
            
            <TableCell>
                <Link
                    to={`/issue/${issue._id}`}
                    style={{ color: "inherit"}} underline="hover"
                >
                    <BuildIcon/>
                </Link>
            </TableCell>
        </TableRow>
    )
}

export const UserTicketsTable = ({ issues, projects}) => {
    if(!issues.length) {
        return (
            <Box
                sx={{
                    display: "flex",
                    flexDirection: "column",
                    alignItems: "center",
                    justifyContent: "center"
                }}
            >
                <Typography variant="h5"> No issues assigned yet!</Typography>
            </Box>
        )
    } else {
        return (
            <TableContainer sx={{height: 350, backgroundColor: "#FFE4B5"}}>
                <Table sx={{ backgroundColor: "#FFE4B5"}}>
                    <TableHead sx={{backgroundColor: "#F4A460"}}>
                        <TableRow>
                            <TableCell align="center" colSpan={5}>
                                USER TICKETS
                            </TableCell>
                        </TableRow>
                        <TableRow>
                            <TableCell align="left">TITLE</TableCell>
                            <TableCell align="left">PRIORITY</TableCell>
                            <TableCell align="left">TYPE</TableCell>
                            <TableCell align="left">PROJECT</TableCell>
                            <TableCell align="left">LINK</TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody sx={{ backgroundColor: "#FFE4B5"}}>
                        {issues.filter(i => i.status !== "solved").map((issue, index) => {
                            const issueProject = projects.find(project => project._id === issue.project);
                            return (
                                <Row key={index} issue={issue} project={issueProject} />
                            )
                        })}
                    </TableBody>
                </Table>
            </TableContainer>
        )
    }
}
