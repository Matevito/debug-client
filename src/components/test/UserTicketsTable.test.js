import { MemoryRouter } from "react-router-dom";
// testing comps
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'

// tested component
import { UserTicketsTable } from "../UserTicketsTable";

describe("UserTicketsTable component", () => {
    
    test("renders a correct table", () => {
        const testIssues = [
            {
                project: "project1",
                title: "issue 1",
                priority: "low",
                type: "feature req",
                _id: "1"
            },
            {
                project: "project1",
                title: "issue 2",
                priority: "high",
                type: "feature req",
                _id: "2"
            }, 
            {
                project: "project2",
                title: "issue 3",
                priority: "mid",
                type: "documentation req",
                _id: "3"
            },
            {
                project: "project1",
                title: "issue 4",
                priority: "mid",
                type: "feature req",
                _id: "4"
            }
        ];
        const TestProjects = [
            {
                _id: "project1",
                title: "project 1"
            },
            {
                _id: "project2",
                title: "project 2"
            }
        ];
        const {container } = render(<UserTicketsTable issues={testIssues} projects={TestProjects} />, {wrapper: MemoryRouter})
        expect(container).toMatchSnapshot()
    })
})