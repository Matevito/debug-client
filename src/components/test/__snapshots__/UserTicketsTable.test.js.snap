// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTicketsTable component renders a correct table 1`] = `
<div>
  <div
    class="MuiTableContainer-root css-1ofbc68-MuiTableContainer-root"
  >
    <table
      class="MuiTable-root css-1w3qztj-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-1lhsthh-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1ndpvdd-MuiTableCell-root"
            colspan="5"
            scope="col"
          >
            USER TICKETS
          </th>
        </tr>
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            TITLE
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            PRIORITY
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            TYPE
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            PROJECT
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            LINK
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-n1y5x-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            issue 1
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-13q6wau-MuiTableCell-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              low
            </p>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            feature req
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <a
              href="/project/project1"
              style="text-decoration: none;"
            >
              project 1
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <a
              href="/issue/1"
              underline="hover"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BuildIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m22.7 19-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
                />
              </svg>
            </a>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            issue 2
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-t8kkpv-MuiTableCell-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              high
            </p>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            feature req
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <a
              href="/project/project1"
              style="text-decoration: none;"
            >
              project 1
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <a
              href="/issue/2"
              underline="hover"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BuildIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m22.7 19-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
                />
              </svg>
            </a>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            issue 3
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1syk86k-MuiTableCell-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              mid
            </p>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            documentation req
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <a
              href="/project/project2"
              style="text-decoration: none;"
            >
              project 2
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <a
              href="/issue/3"
              underline="hover"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BuildIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m22.7 19-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
                />
              </svg>
            </a>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            issue 4
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1syk86k-MuiTableCell-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              mid
            </p>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            feature req
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <a
              href="/project/project1"
              style="text-decoration: none;"
            >
              project 1
            </a>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <a
              href="/issue/4"
              underline="hover"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BuildIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m22.7 19-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
                />
              </svg>
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
